Could not build 2.0.0-beta3 on CentOS 7

Env:
* CentOS Linux release 7.2.1511 (Core)
* gcc version 4.8.5 20150623 (Red Hat 4.8.5-4) (GCC)

Build Script:
* [Dockerfile](https://github.com/fetus-hina/h2o-rpm/blob/next/Dockerfile.centos7)
* [SPEC file](https://github.com/fetus-hina/h2o-rpm/blob/next/rpmbuild/SPECS/h2o.spec)

Error:
```
[ 58%] Built target h2o
Scanning dependencies of target libh2o-evloop
[ 58%] Building C object CMakeFiles/libh2o-evloop.dir/deps/cloexec/cloexec.c.o
[ 58%] Building C object CMakeFiles/libh2o-evloop.dir/deps/libyrmcds/close.c.o
[ 58%] Building C object CMakeFiles/libh2o-evloop.dir/deps/libyrmcds/connect.c.o
[ 61%] Building C object CMakeFiles/libh2o-evloop.dir/deps/libyrmcds/recv.c.o
[ 61%] Building C object CMakeFiles/libh2o-evloop.dir/deps/libyrmcds/send.c.o
[ 61%] Building C object CMakeFiles/libh2o-evloop.dir/deps/libyrmcds/send_text.c.o
[ 61%] Building C object CMakeFiles/libh2o-evloop.dir/deps/libyrmcds/socket.c.o
[ 64%] Building C object CMakeFiles/libh2o-evloop.dir/deps/libyrmcds/strerror.c.o
[ 64%] Building C object CMakeFiles/libh2o-evloop.dir/deps/libyrmcds/text_mode.c.o
[ 64%] Building C object CMakeFiles/libh2o-evloop.dir/deps/picohttpparser/picohttpparser.c.o
[ 64%] Building C object CMakeFiles/libh2o-evloop.dir/lib/common/file.c.o
[ 64%] Building C object CMakeFiles/libh2o-evloop.dir/lib/common/filecache.c.o
[ 66%] Building C object CMakeFiles/libh2o-evloop.dir/lib/common/hostinfo.c.o
[ 66%] Building C object CMakeFiles/libh2o-evloop.dir/lib/common/http1client.c.o
[ 66%] Building C object CMakeFiles/libh2o-evloop.dir/lib/common/memcached.c.o
[ 66%] Building C object CMakeFiles/libh2o-evloop.dir/lib/common/memory.c.o
[ 69%] Building C object CMakeFiles/libh2o-evloop.dir/lib/common/multithread.c.o
[ 69%] Building C object CMakeFiles/libh2o-evloop.dir/lib/common/serverutil.c.o
[ 69%] Building C object CMakeFiles/libh2o-evloop.dir/lib/common/socket.c.o
In file included from /usr/include/openssl/kssl.h:71:0,
                 from /usr/include/openssl/ssl.h:165,
                 from //rpmbuild/BUILD/h2o-2.0.0-beta3/include/h2o/socket.h:31,
                 from //rpmbuild/BUILD/h2o-2.0.0-beta3/lib/common/socket.c:31:
//rpmbuild/BUILD/h2o-2.0.0-beta3/lib/common/../../deps/ssl-conservatory/openssl/openssl_hostname_validation.c:23:12: error: expected identifier or '(' before '__extension__'
 static int tolower(int ch) {
            ^
make[2]: *** [CMakeFiles/libh2o-evloop.dir/lib/common/socket.c.o] Error 1
make[1]: *** [CMakeFiles/libh2o-evloop.dir/all] Error 2
make: *** [all] Error 2
error: Bad exit status from /var/tmp/rpm-tmp.sBRx1a (%build)
```
The function `tolower()` looks defined other header file `/usr/include/ctype.h`.
I didn't verify yet about including path and how rewrite the source by CPP.

I checked out h2o branch of `kazuho/ssl-conservatory`, and `make` on its `openssl` directory.
Then, I got error below:
```
gcc -c -Wall -std=c99 -pedantic   -c -o openssl_hostname_validation.o openssl_hostname_validation.c
openssl_hostname_validation.c:23:12: error: static declaration of ‘tolower’ follows non-static declaration
 static int tolower(int ch) {
            ^
make: *** [openssl_hostname_validation.o] Error 1
```

And if `make` on its master branch, looks built it successfully.

あまり伝わらない気がしますので、日本語でも書きます。

2.0.0-beta3 を CentOS 7 でビルドしようとしたところ、上述のようなエラーが発生しました。
ssl-conservatory の `tolower()` でソース上は見えない `__extension__` あたりが気にくわないようです。
ざっくり検索したところ、 `/usr/include/ctype.h` の中に `#define` があり `tolower` トークンが置換されるパスがあるように見えます（実際にどのような include パスが構成され、どう置換されているかは（まだ）確認していません。この置換の中に `__extension__` というトークンは見えます）。

`kazuho/ssl-conservatory` の `h2o` ブランチをチェックアウトして openssl ディレクトリの中で単に make してみると、単にこれだけでもエラーは発生するようです（エラーの内容は異なります）。
master ブランチを make すると、（tolower が無いので当然ではありますが）正常にビルドできるように見えます（エラーが出ずmakeが終了することだけしか確認はしていません）。

--- a/deps/ssl-conservatory/openssl/openssl_hostname_validation.c
+++ b/deps/ssl-conservatory/openssl/openssl_hostname_validation.c
@@ -20,7 +20,7 @@
 
 #define HOSTNAME_MAX_SIZE 255
 
-static int tolower(int ch) {
+static int lowercase(int ch) {
 	if ('A' <= ch && ch <= 'Z')
 		return ch - 'A' + 'a';
 	return ch;
@@ -30,7 +30,7 @@ static int memeq_ncase(const char *x, const char *y, size_t l) {
 	if (l == 0)
 		return 1;
 	do {
-		if (tolower(*x++) != tolower(*y++))
+		if (lowercase(*x++) != lowercase(*y++))
 			return 0;
 	} while (--l != 0);
 	return 1;

